src/main.ts

```diff
@@ -388,95 +433,66 @@ export default class ReadwisePlugin extends Plugin {
     let booksToRefresh = this.settings.booksToRefresh;
     booksToRefresh.push(bookId);
+    console.log(`Readwise Official plugin: added book id ${bookId} to refresh later`);
     this.settings.booksToRefresh = booksToRefresh;
     await this.saveSettings();
   }
 
-  reimportFile(vault: Vault, fileName: string) {
-    const bookId = this.settings.booksIDsMap[fileName];
+  async removeBooksFromRefresh(bookIds: Array<string> = []) {
+    if (!bookIds.length) return;
+
+    console.log(`Readwise Official plugin: removing book ids ${bookIds.join(', ')} from refresh list`);
+    this.settings.booksToRefresh = this.settings.booksToRefresh.filter(n => !bookIds.includes(n));
+    await this.saveSettings();
+  }
+
+  async reimportFile(vault: Vault, fileName: string) {
     try {
-      fetch(
-        `${baseURL}/api/refresh_book_export`,
-        {
-          headers: {...this.getAuthHeaders(), 'Content-Type': 'application/json'},
-          method: "POST",
-          body: JSON.stringify({exportTarget: 'obsidian', books: [bookId]})
-        }
-      ).then(response => {
-        if (response && response.ok) {
-          let booksToRefresh = this.settings.booksToRefresh;
-          this.settings.booksToRefresh = booksToRefresh.filter(n => ![bookId].includes(n));
-          this.saveSettings();
-          vault.delete(vault.getAbstractFileByPath(fileName));
-          this.startSync();
-        } else {
-          this.notice("Failed to reimport. Please try again", true);
-        }
-      });
+      this.notice("Deleting and reimporting file...", true);
+      await vault.delete(vault.getAbstractFileByPath(fileName));
+      const bookId = this.settings.booksIDsMap[fileName];
+      await this.refreshBookExport([bookId]);
     } catch (e) {
       console.log("Readwise Official plugin: fetch failed in Reimport current file: ", e);
     }
   }
 
-  startSync() {
+  async startSync() {
     if (this.settings.isSyncing) {
       this.notice("Readwise sync already in progress", true);
     } else {
       this.settings.isSyncing = true;
-      this.saveSettings();
-      this.requestArchive();
+      await this.saveSettings();
+      await this.requestArchive();
     }
     console.log("Readwise Official plugin: started sync");
   }
 
   async onload() {
-    // @ts-ignore
+    await this.loadSettings();
+
+    // @ts-expect-error - no type for isMobile
     if (!this.app.isMobile) {
       this.statusBar = new StatusBar(this.addStatusBarItem());
       this.registerInterval(
         window.setInterval(() => this.statusBar.display(), 1000)
       );
     }
-    await this.loadSettings();
-    this.refreshBookExport = debounce(
-      this.refreshBookExport.bind(this),
-      800,
-      true
-    );
-
-    this.refreshBookExport(this.settings.booksToRefresh);
-    this.app.vault.on("delete", async (file) => {
-      const bookId = this.settings.booksIDsMap[file.path];
-      if (bookId) {
-        await this.addBookToRefresh(bookId);

```

Comment:
I think this is the only place where we use `addBookToRefresh` so if this goes, the function should also go, albeit it's not yet clear to me we can remove it fully...

Reply:
`addBookToRefresh` is used `catch` of in `downloadArchive` when it fails to write.

In theory... the book ID may not exist when the `catch` is triggered as the book ID is set partway through the `try`.

So it's possible that we could still probably be purged if we pull the name/bookID splitting up out of the `try/catch` and then save the book ID to the settings. In fact, I think it's probably a good idea to refactor this part of the code, otherwise you might end up with `undefined` in the `booksToRefresh`.

I'll investigate this after more critical parts of feedback.

Reply:
OK, so I don't think `addBookToRefresh` can be eliminated because it covers the case where the user **doesn't** have "resync deleted books" enabled **AND** an entry couldn't be saved to the filesystem.

That said, I still reworked the code a little:
- 62f2744 (logic rework)
- bd6f30c (add comment about use of `addBookToRefresh`)